### PR TITLE
SyncBot: Improve resilience of Front linked thread searching

### PR DESCRIPTION
**Pull request by sqweelygig**
* Increase the uniqueness of each `duff` string
* Fixed a comment I happened to notice was wrong
* Added error messages for Flowdock 5xx errors

This is because increasing the error output revealed that SyncBot is attempting to post messages to a duff thread id on Flowdock

Change-Type: patch
Connects-To: #367 

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [x] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #367

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +47,7 @@
     "@types/request-promise": "^4.1.36",
     "chai": "^4.1.2",
     "chai-as-promised": "^7.1.1",
+    "fs-readfile-promise": "^3.0.1",
```
Comment (hedss) [inline package.json]: See comment in testfile as to why I don't think this is required.

File: tests/services/messenger/translators/front.spec.ts [hunk 2]
```diff
@@ -1,10 +1,48 @@
 /// <reference types="mocha" />
 import { expect } from 'chai';
 
+import * as readFile from 'fs-readfile-promise';
```
Comment (hedss) [inline tests/services/messenger/translators/front.spec.ts]: If you're just using it for reading a file:
```
import * as FS from 'fs';
const readFile = Promise.promisify(FS.readFile);
```
I'm all in favour of utilities, but this seems extremely heavy for just reading a file when an extra line will get you exactly the same results.

Maybe we should be including a utility service for general FS access.

File: lib/bots/syncbot.ts [hunk 3]
```diff
@@ -269,8 +269,8 @@ export class SyncBot extends ProcBot {
 				title: data.details.title,
 			},
 			source: {
-				message: 'duff',
-				thread: 'duff',
+				message: 'duff_SyncBot_createErrorComment_a',
```
Comment (hedss) [inline lib/bots/syncbot.ts]: Just a comment currently, but maybe we should add in future something that allows the creation of a unique set of identifier messages, something that texts a text string then appends line number to it?

Comment (sqweelygig) [reply lib/bots/syncbot.ts]: Good shout.  Will add an issue.

**Pull request merged**